test/lib/code_corps/validators/slug_validator_test.exs

```diff
@@ -0,0 +1,111 @@
+defmodule CodeCorps.Validators.SlugValidatorTest do
+  use ExUnit.Case, async: true
+
+  import CodeCorps.Validators.SlugValidator
+
+  test "with only letters" do
+    changeset = cast_slug("testslug") # can't be `slug` because reserved
+    changeset = validate_slug(changeset, :slug)
+    assert changeset.valid?
+  end
+
+  test "with preceding underscores" do
+    changeset = cast_slug("_slug")
+    changeset = validate_slug(changeset, :slug)
+    assert changeset.valid?
+  end
+
+  test "with suffixed underscores" do
+    changeset = cast_slug("slug_")
+    changeset = validate_slug(changeset, :slug)
+    assert changeset.valid?
+  end
+
+  test "with preceding numbers" do
+    changeset = cast_slug("123slug")
+    changeset = validate_slug(changeset, :slug)
+    assert changeset.valid?
+  end
+
+  test "with suffixed numbers" do
+    changeset = cast_slug("slug123")
+    changeset = validate_slug(changeset, :slug)
+    assert changeset.valid?
+  end
+
+  test "with multiple dashes" do
```

Comment:
Eh actually up to debate since it looks like you're separating the valid and non-valid ones